### ISSUE TITLE
OPL3 Duo! playback corruption in Linux

### ISSUE THREAD

**Issue by zomgugoff** [joncampbell123/dosbox-x#2995]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar bug report(s) already exists?

- [X] I have searched and didn't find any similar issues.

### What operating system(s) this bug have occurred on?

Arch Linux

### What version(s) of DOSBox-X have this bug?

0.83.18 both SDL1 and 2

### Describe the bug

The playback from the OPL3 Duo! board in Linux is very wrong. Hanging notes, notes not playing, instruments are wrong and distorted. I would guess it's timing issue, but it could be something with communication over USB. The board I am using works fine in Windows.

### Expected behavior

Playback should be accurate

### Steps to reproduce the behaviour

Plug in the OPL3 Duo!

Make sure your user account on the system is in the group that the /dev/ttyACM0 device is (in Arch, it's uucp, but other distros apparently use dialout) so Dosbox-X can access it, or run Dosbox-X with sudo.

Edit the Dosbox-X config file to use OPL3 Duo! with the oplport set to ttyACM0

Run Dosbox-X and make sure it was able to initialize the board successfully:

OPL3 Duo! Board: Connecting to port ttyACM0... 
OPL3 Duo! Board: COM Port OK.

Run any game that uses OPL3. I tested with Commander Keen4 and Budokan.

### Used configuration

```ini
Default apart from these changes:

oplemu                                           = opl3duoboard
oplport                                          = ttyACM0
```

### Emulator log

```text
There are no messages in the log after initialization.

OPL3 Duo! Board: Connecting to port ttyACM0... 
OPL3 Duo! Board: COM Port OK.
```

**Comment by zomgugoff** [comment]
Quick update: This issue was not fixed in 0.83.20

**Comment by KGOrphanides** [comment]
This may require its own issue report, but I'm currently getting similar issues with an OPL2 Audio Board. 
Timing, playback speed, and instrument errors on titles including Quest for Glory II and Elvira, Mistress of the Dark.

OS: Pop!_OS 21.04
DOSBox-X version: 0.83.19

Using  /dev/ttyACM0, user is in the dialout group.

I don't have a Windows PC to compare, but I'll see if I can borrow one.
